src/ring/middleware/gzip.clj

```diff
@@ -60,20 +60,31 @@
                       (dissoc "Content-Length")))
       (update-in [:body] piped-gzipped-input-stream)))
 
-(defn wrap-gzip [handler]
-  (fn [req]
-    (let [{:keys [body status] :as resp} (handler req)]
-      (if (and (= status 200)
-               (not (get-in resp [:headers "Content-Encoding"]))
-               (or
-                (and (string? body) (> (count body) 200))
-                (and (seq? body) @flushable-gzip?)
-                (instance? InputStream body)
-                (instance? File body)))
-        (let [accepts (get-in req [:headers "accept-encoding"] "")
-              match (re-find #"(gzip|\*)(;q=((0|1)(.\d+)?))?" accepts)]
-          (if (and match (not (contains? #{"0" "0.0" "0.00" "0.000"}
-                                         (match 3))))
-            (gzipped-response resp)
-            resp))
-        resp))))
+(defn- wrap-gzip* [req {:keys [body status] :as resp}]
```

Comment:
Can you change this to `gzip-request` please?

Reply:
No problem. Would `gzip-response` make more sense? Because it's the response that's getting zipped, not the request. In the JSON middleware you linked they're going with `json-response` for the function that does the same thing

Reply:
Sorry, yes you're completely right.